### PR TITLE
Add org.tildearrow.furnace

### PR DESCRIPTION
**Pull request by ColinKinloch**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** (hold off until stable 0.6) https://github.com/tildearrow/furnace/pull/1475#issuecomment-1714671760
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

### ToDo
- [x] Fix jack
- ~[ ] Submit jack patch upstream~ (fixed linking issue)
- [x] Ask authors permission to package
- [x] Maybe compile rtmidi and fmt
- [x] Figure out how to access instruments, samples and demo music in `/app/share/furnace` via the open file dialog  (Added explanation in README)
- [x] Drag and drop https://github.com/libsdl-org/SDL/pull/8257

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by ColinKinloch** [comment]
bot, build org.tildearrow.Furnace

**Comment by ColinKinloch** [comment]
bot, build org.tildearrow.Furnace

File: org.tildearrow.Furnace.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+app-id: org.tildearrow.Furnace
```
Comment (hfiguiere) [inline org.tildearrow.Furnace.yml:1]: the appdata say `  <id>org.tildearrow.furnace</id>`

You should is it instead.

**Comment by ColinKinloch** [comment]
bot, build org.tildearrow.Furnace

**Comment by ColinKinloch** [comment]
bot, build org.tildearrow.furnace

**Comment by ColinKinloch** [comment]
bot, build org.tildearrow.furnace

**Comment by ColinKinloch** [comment]
bot, build org.tildearrow.furnace

**Comment by ColinKinloch** [comment]
bot, build org.tildearrow.furnace

**Comment by ColinKinloch** [comment]
bot, build org.tildearrow.furnace

**Comment by ColinKinloch** [comment]
bot, build org.tildearrow.furnace

File: org.tildearrow.furnace.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --device=dri
+  - --socket=pulseaudio
+  - --filesystem=xdg-run/pipewire-0
+
+modules:
+  - shared-modules/SDL2/SDL2-with-libdecor.json
```
Comment (bbhtt) [inline org.tildearrow.furnace.yml:19]: 23.08 has this https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/1533f8d0393984f5684985c8b710e9fbaf9c8734/elements/components/sdl2.bst#L12, https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/1533f8d0393984f5684985c8b710e9fbaf9c8734/elements/components/sdl2.bst#L32

**Comment by ColinKinloch** [comment]
What's the best way to provide access to data in `/app/share/furnace` (demos/instruments/samples), since the file open dialogue in furnace uses portals and it displays the host filesystem.

During testing I've been running:
```
flatpak run --command=sh org.tildearrow.furnace//test -c 'gio open /app/share/furnace'
```

**Comment by hfiguiere** [comment]
> What's the best way to provide access to data in `/app/share/furnace` (demos/instruments/samples), since the file open dialogue in furnace uses portals and it displays the host filesystem.

You can't.

**Comment by ColinKinloch** [comment]
bot, build org.tildearrow.furnace

**Comment by ColinKinloch** [comment]
bot, build org.tildearrow.furnace

File: org.tildearrow.furnace.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      - type: patch
+        path: dnd_portal.patch
+    cleanup:
+      - /bin/sdl2-config
+      - /include
+      - /lib/libSDL2.la
+      - /lib/libSDL2main.a
+      - /lib/libSDL2main.la
+      - /lib/libSDL2_test.a
+      - /lib/libSDL2_test.la
```
Comment (bbhtt) [inline org.tildearrow.furnace.yml:34]:
```suggestion
      - '*.a'
      - '*.la'
```

File: dnd_portal.patch
Comment (bbhtt) [inline dnd_portal.patch:1]: I wonder if you can upstream this?

Comment (ColinKinloch) [reply dnd_portal.patch:1]: I did :) https://github.com/libsdl-org/SDL/pull/8257

**Comment by ColinKinloch** [comment]
bot, build org.tildearrow.furnace

File: dnd_portal.patch
Comment (bbhtt) [reply dnd_portal.patch:1]: Awesome!

**Comment by ColinKinloch** [comment]
bot, build org.tildearrow.furnace

**Pull request closed without merging**